### PR TITLE
feat: add `beforeEachBundleCommand`

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by FabianLars** [comment]
What's the use case for this? Or rather, what's the issue with the current beforeBundleCommand?

**Comment by amrbashir** [comment]
This is allows for some conditional logic based on which package type is being bundled.

Also maybe we should also add `beforeEachTargetBuildCommand`? but I don't see a use-case for it immediately, and probably will only affect when building for Android as it has more than one target being built in the same invoke of `tauri android build`?

I don't really think this PR is urgent or needed that much so I am fine to just keep it open until someone requests it or close it.

**Comment by FabianLars** [comment]
Ahh okay, my bad, i missed the `TAURI_BUNDLER_PACKAGE_TYPE` at the bottom (it's missing in the config description). Totally get it then 👍

**Comment by amrbashir** [comment]
good catch, I will add it to the docs

File: core/tauri-utils/src/config.rs [hunk 1]
```diff
@@ -1712,6 +1712,12 @@ pub struct BuildConfig {
   /// The TAURI_PLATFORM, TAURI_ARCH, TAURI_FAMILY, TAURI_PLATFORM_VERSION, TAURI_PLATFORM_TYPE and TAURI_DEBUG environment variables are set if you perform conditional compilation.
   #[serde(alias = "before-bundle-command")]
   pub before_bundle_command: Option<HookCommand>,
+  /// Similar to `before_bundle_command`, a shell command to run before the bundling phase
+  /// of each bundle format.
```
Comment (amrbashir) [inline core/tauri-utils/src/config.rs]:
```suggestion
  /// of each package type. Check `TAURI_BUNDLER_PACKAGE_TYPE` env var to know
  /// which package type is being bundled.
```

**Comment by amrbashir** [comment]
Closing until we see a use-case for this

**Pull request closed without merging**